packages/hooks/src/createRunHook.js

```diff
@@ -30,6 +30,14 @@ function createRunHook( hooks, returnFirstArg ) {
 
 		const handlers = hooks[ hookName ].handlers;
 
+		// The following code can be stripped from production builds.
+		/* develblock:start */
```

Comment:
Is this meant to be from a Webpack plugin we're not using?

https://www.npmjs.com/package/webpack-strip-block

We may consider instead a combination of Webpack's (built-in) DefinePlugin and dead-code elimination from its (built-in) minification:

https://webpack.js.org/plugins/define-plugin/

Prior art:

- https://github.com/WordPress/gutenberg/blob/31b6b175af43862dbdcfb9a05ca4887f3ad60116/packages/blocks/src/api/validation.js#L257-L263
- #13324

Reply:
very interesting.

i was actually imagining that the all hook handling would remain in the package by default and that we would _strip it in the WordPress core build for the minified file_. 

In core we already build two versions of each script: one for SCRIPT_DEBUG and one for production. We can use these tags and the webpack-strip-block plugin to remove them in core's production build.  This would completely eliminate the code for produciton and not introduce the overhead of checking the constant.

What do you think?

Reply:
The treatment of the environment variable is a bit of a facade, really. It actually achieves the same effect as what you're describing. The way the DefinePlugin works is to literally swap the identifier with some substitute. Thus, if you introduced a condition like this, the body of the condition would be kept for development, and removed for production.

Before (source):

```js
/* a */
if ( process.env.NODE_ENV !== 'production' ) {
	/* b */
}
/* c */
```

After (built with Webpack, unminified):

```js
/* a */
if ( 'production' !== 'production' ) {
	/* b */
}
/* c */
```

After (built with Webpack, minified, comments kept for demonstration only):

```js
/* a *//* c */
```

Reply:
Ok, cool - so we don't need the or the tagging, just a conditional, got it. thanks for clarifying.

So in terms of what we ship in the package, we only ship source right? Do we ship debug scripts with the GB plugin or in core?

my ultimate goal is to have this depend of a wp-config constant, ideally SCRIPT_DEBUG. Do you think its fine keeping this in the pa

Reply:
> So in terms of what we ship in the package, we only ship source right? Do we ship debug scripts with the GB plugin or in core?

My expectation would have been that `process.env.NODE_ENV` would be effectively "development" for the non-minified scripts we ship with core, and thus if `SCRIPT_DEBUG` would cause the unminified form to load, it would take effect as you expect. However, considering [an example](https://github.com/WordPress/gutenberg/blob/910b655018037167c2dcee38095905003f632b51/packages/components/src/tooltip/index.js#L94-L97) where we `console.error` a message, I see no reference to said message in the [full unminified source](https://raw.githubusercontent.com/WordPress/WordPress/master/wp-includes/js/dist/components.js) for `wp.components`. I'm not too sure if this is intended or not.

Reply:
I think that `process.env.NODE_ENV` would work the way it's expected. WordPress core would have to generate two versions of the files: `development` and `production`. `SCRIPT_DEBUG` would pick the proper version then. If this isn't the case in core, we should fix it.